admin/app/components/solidus_admin/ui/modal/component.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class SolidusAdmin::UI::Modal::Component < SolidusAdmin::BaseComponent
+  renders_one :body
+  renders_one :actions
+
+  def initialize(
+    title:,
+    id: SecureRandom.hex,
+    close_message: "Close",
+    button_component: component("ui/button")
+  )
+    @id = id
+    @title = title
+    @close_message = close_message
+    @button_component = button_component
+  end
+
+  def render_cancel_button
+    render @button_component.new(text: @close_message, scheme: :secondary, class: 'sm:mt-0 sm:w-auto', 'data-action': "#{stimulus_id}#toggleModal", "data-#{stimulus_id}-modal-id-param": @id)
+  end
```

Comment:
Could we inline the close button rendering in the `component.html.erb` file instead of using a separate method?
Given its simplicity, keeping it in the main template could enhance readability at this stage 🤔